Model/Api/ShippingMethods.php

```diff
@@ -595,7 +595,28 @@ public function shippingEstimation($quote, $addressData)
             if ($serialized = $this->cache->load($cacheIdentifier)) {
                 $address = $quote->isVirtual() ? $quote->getBillingAddress() : $quote->getShippingAddress();
                 $address->setShippingMethod(null)->save();
-                return $this->serialize->unserialize($serialized);
+                try {
+                    $shippingOptionsData = $this->serialize->unserialize($serialized);
+                    //re-create the object expected as the return
+                    return $this->shippingOptionsInterfaceFactory->create()
```

Comment:
Could you please explain this code?

Reply:
I got it. Could you move this code to the helper like createShippingOptionsByArray ?

Reply:
done